### PR TITLE
[feat]: [IAC-1222]: Add multiple steps for multiples repositories in IACM

### PR DESCRIPTION
**Pull request by nassergonzalez**
## Describe your changes

| Use case | Workspace 1 branch | Workspace 2 branch | Trigger | TF file workspace 1 branch | TF file workspace 2 branch | Cloned | Workspace 1 cloned branch | Workspace 2 cloned branch | TF file cloned branch 1 | TF file cloned branch 2 |
|-|-|-|-|-|-|-|-|-|-|-|  
| 1 | repo1 branch main | repo2 branch main | Manual | x | x |CLONED| repo 1 branch main | repo 2 branch main | x | x |
| 2 | repo1 branch main | repo2 branch main | PR repo 1 branch foo | x | x |CLONED| repo 1 branch foo | repo 2 branch main | x | x |
| 3 | repo1 branch main | repo2 branch main | PR repo 2 branch bar | x | x |CLONED| repo 1 branch main | repo 2 branch bar | x | x |  
| 4 | repo1 branch main | repo2 branch main | Push repo 1 branch main | x | x |CLONED| repo 1 branch main | repo 2 branch main | x | x |
| 5 | repo1 branch main | repo2 branch main | Push repo 2 branch main | x | x |CLONED| repo 1 branch main | repo 2 branch main | x | x |
| 6 | repo1 branch main | repo2 branch main | Manual | repo3 branch main | repo4 branch main |CLONED| repo 1 branch main | repo 2 branch main | repo 3 branch main | repo 4 branch main |
| 7 | repo1 branch main | repo2 branch main | Trigger repo 1 branch foo | repo3 branch main | repo4 branch main |CLONED| repo 1 branch foo | repo 2 branch main | repo 3 branch main | repo 4 branch main |
| 8 | repo1 branch main | repo2 branch main | Trigger repo 2 branch bar | repo3 branch main | repo4 branch main |CLONED| repo 1 branch main | repo 2 branch bar | repo 3 branch main | repo 4 branch main |
| 9 | repo1 branch main | repo2 branch main | Trigger repo 3 branch zar | repo3 branch main | repo4 branch main |CLONED| repo 1 branch main | repo 2 branch main | repo 3 branch zar | repo 4 branch main |
| 10 | repo1 branch main | repo2 branch main | Trigger repo 4 branch far | repo3 branch main | repo4 branch main |CLONED| repo 1 branch main | repo 2 branch main | repo 3 branch main | repo 4 branch far |
| 11 | repo1 branch main | x | Trigger repo 1 branch far->main | repo1 branch lalalal | x |CLONED| repo 1 branch far | x| x| x |
| 12 | repo1 branch main |x| Trigger repo 1 branch lalala | repo3 branch lalala | x |CLONED| x | x | repo3 lalala| x |

NOT SUPPORTED
| Use case | Workspace 1 branch | Workspace 2 branch | Trigger | TF file workspace 1 branch | TF file workspace 2 branch | Cloned | Workspace 1 cloned branch | Workspace 2 cloned branch | TF file cloned branch 1 | TF file cloned branch 2 |
|-|-|-|-|-|-|-|-|-|-|-|  
| 1 | repo1 branch main |x| Manual | repo1 branch lalala | x |CLONED| x | x | repo1 main x |
| 2 | repo1 branch main |repo 1 branch lalala|  |  | x |CLONED| repo 1 branch main | x | x| x |

If the table is not readable you have the same thing here:

image

## Checklist
- [ ] I've documented the changes in the PR description.
- [ ] I've tested this change either in PR or local environment.
- [ ] If hashcheck failed I followed [the checklist](https://harness.atlassian.net/wiki/spaces/DEL/pages/21016838831/PR+Codebasehash+Check+merge+checklist) and updated this PR description with my answers to make sure I'm not introducing any breaking changes.

## Comment Triggers
<details>
  <summary>Build triggers</summary>

- Feature build: `trigger feature-build`
- 
  Specific builds
  - `trigger manager`
  - `trigger dms`
  - `trigger ng_manager`
  - `trigger cvng `
  - `trigger cmdlib`
  - `trigger template_svc`
  - `trigger events_fmwrk_monitor`
  - `trigger event_server`
  - `trigger change_data_capture`
  -  Trigger multiple builds together: For eg: `trigger dms, manager`
- Immutable delegate `trigger publish-delegate`
</details>

## [Latest PR Check Triggers](https://github.com/harness/harness-core/blob/develop/.github/pull_request_template.md)

<details>
  <summary>PR Check triggers</summary>
You can run multiple PR check triggers by comma separating them in a single comment. e.g. `trigger ut0, ut1`

- Compile: `trigger compile`
- CodeformatCheckstyle: `trigger checkstylecodeformat`
  - CodeFormat: `trigger codeformat`
  - Checkstyle: `trigger checkstyle`
- MessageMetadata: `trigger messagecheck`
- File-Permission-Check: `trigger checkpermission`
- Recency: `trigger recency`
- BuildNumberMetadata: `trigger buildnum`
- Trigger CommonChecks: `trigger commonchecks`
- PMD: `trigger pmd`
- Copyright Check: `trigger copyrightcheck`
- Feature Name Check: `trigger featurenamecheck`
- UnitTests-ALL: `trigger uts`
- UnitTests-0: `trigger ut0`
- UnitTests-1: `trigger ut1`
- UnitTests-2: `trigger ut2`
- UnitTests-3: `trigger ut3`
- UnitTests-4: `trigger ut4`
- UnitTests-5: `trigger ut5`
- UnitTests-6: `trigger ut6`
- UnitTests-7: `trigger ut7`
- UnitTests-8: `trigger ut8`
- UnitTests-9: `trigger ut9`
- CodeBaseHash: `trigger codebasehash`
- CodeFormatCheckstyle: `trigger checkstylecodeformat`
- SonarScan: `trigger ss`
- GitLeaks: `trigger gitleaks`
- Trigger all Checks: `trigger smartchecks`
- Go Build: `trigger gobuild`
- Validate_Reviews: `trigger review`
- Module Dependency Check: `trigger mdc`
</details>

## PR check failures and solutions
https://harness.atlassian.net/wiki/spaces/BT/pages/21106884744/PR+Checks+-+Failures+and+Solutions

### Rebase Instructions:
To rebase your branch onto the latest changes in the target branch, simply use the following command in a comment: `/rebase`

## [Contributor license agreement](https://github.com/harness/harness-core/blob/develop/CONTRIBUTOR_LICENSE_AGREEMENT.md)

- - -
This change is [Reviewable](https://reviewable.io/reviews/harness/harness-core/56281)

### CODE REVIEW

File: 310-iacm-manager/app/src/main/java/io/harness/iacm/plan/creator/stage/IACMStagePMSPlanCreator.java [hunk 1]
```diff
@@ -188,59 +200,55 @@ public LinkedHashMap<String, PlanCreationResponse> createPlanForChildrenNodes(
...
         buildObject.spec(TagBuildSpec.builder()
-                             .tag(ParameterField.<String>builder().value(variablesRepo.getRepository_commit()).build())
+                             .tag(ParameterField.<String>builder().value(workspace.getRepository_commit()).build())
                              .build());
       }
-      String hashedGitRepoInfo = iacmStepsUtils.generateHashedGitRepoInfo(variablesRepo.getRepository(),
-          variablesRepo.getRepository_connector(), variablesRepo.getRepository_branch(),
-          variablesRepo.getRepository_commit(), variablesRepo.getRepository_path());
+
+      String cloneVar = "/harness/";
```
Comment (scottyw-harness) [inline 310-iacm-manager/app/src/main/java/io/harness/iacm/plan/creator/stage/IACMStagePMSPlanCreator.java:238]: This maybe won't work for Docker delegates?

**Review by scottyw-harness — DISMISSED** [review]

**Review by harshjain12 — APPROVED** [review]

**Comment by nassergonzalez** [comment]
trigger uts

**Comment by nassergonzalez** [comment]
trigger recency /rebase

**Comment by nassergonzalez** [comment]
trigger uts

**Pull request merged**